### PR TITLE
Hotfix/refund amount

### PR DESCRIPTION
**Pull request by tomrndom**
* Uses amounts from order on for purchased orders
* Uses calculations to show the order detail on the purchase flow

### CODE REVIEW

File: src/components/order-summary.js [hunk 1]
```diff
@@ -52,29 +53,23 @@ class OrderSummary extends React.Component {
                   ticketSummary.push({ticket_type_id: (tix.type_id ? tix.type_id : tix.ticket_type_id), tix_type: tixType, name, qty: 1})
               }
   
-              ticketTotal = ticketTotal + tixType.cost;
+              purchaseTicketTotal = purchaseTicketTotal + tixType.cost;
```
Comment (smarcet) [inline src/components/order-summary.js:56]: the ticket type cost could change on a later stage
can we use the ticket cost ?
on an ongoing purchase do you have that info ?
regards

**Review by smarcet — CHANGES_REQUESTED** [review]
please review comments

File: src/components/order-summary.js (hunk 1 above)
Comment (tomrndom) [reply src/components/order-summary.js:56]: The data is available after the reservation is made (step 3), but not on the second step.

Also the step 3 is displaying the due total incorrectly, and not showing the taxes for the order. Currently working on it

File: src/components/order-summary.js [hunk 2]
```diff
@@ -101,29 +84,39 @@ class OrderSummary extends React.Component {
...
 
-                        {discountTotal > 0 &&
+                        {reservation?.discount_amount > 0 || discount_amount > 0 &&
                         <div className="row order-discounts order-row">
                             <div className="col-xs-7 text-left">
                                 {T.translate("order_summary.discounts")}
                             </div>
                             <div className="col-xs-5 text-right subtotal">
-                                -${discountTotal.toFixed(2)}
+                                -${reservation?.discount_amount ? reservation.discount_amount.toFixed(2) : discount_amount?.toFixed(2)}
```
Comment (smarcet) [inline src/components/order-summary.js:99]: could you merge this 2 variables on one before the return
like 
const discountAmount = reservation?.discount_amount ? reservation.discount_amount.toFixed(2) : discount_amount?.toFixed(2);

File: src/components/order-summary.js [hunk 3]
```diff
@@ -101,29 +84,39 @@ class OrderSummary extends React.Component {
...
                         }
-                        {refundTotal > 0 && 
+                        {reservation?.refunded_amount > 0 || refunded_amount > 0 && 
                         <div className="row order-refunds order-row">
                             <div className="col-xs-7 text-left">
                                 {T.translate("order_summary.refunds")}                          
                             </div>
                             <div className="col-xs-5 text-right subtotal">
-                                -${refundTotal.toFixed(2)}
+                                -${reservation?.refunded_amount ? reservation.refunded_amount.toFixed(2) : refunded_amount?.toFixed(2)}
```
Comment (smarcet) [inline src/components/order-summary.js:109]: same as https://github.com/fntechgit/summit-registration/pull/60/files#r714355968

File: src/components/order-summary.js [hunk 4]
```diff
@@ -101,29 +84,39 @@ class OrderSummary extends React.Component {
...
+                            </div>
+                        </div>
+                        }
+                        {reservation?.taxes_amount > 0 || taxes_amount > 0 && 
+                        <div className="row order-refunds order-row">
+                            <div className="col-xs-7 text-left">
+                                {T.translate("order_summary.taxes")}
+                            </div>
+                            <div className="col-xs-5 text-right subtotal">
+                                ${reservation?.taxes_amount? reservation.taxes_amount.toFixed(2) : taxes_amount?.toFixed(2)}
```
Comment (smarcet) [inline src/components/order-summary.js:119]: same as https://github.com/fntechgit/summit-registration/pull/60/files#r714355968

File: src/components/order-summary.js [hunk 5]
```diff
@@ -143,12 +136,11 @@ class OrderSummary extends React.Component {
                                 {T.translate("order_summary.total")}
                             </div>
                             <div className="col-xs-6 text-right total">
-                                ${ order.status === 'Paid' ? '0.00' : total.toFixed(2)}
+                                ${ order.status === 'Paid' ? '0.00' : reservation?.amount? reservation.amount.toFixed(2) : amount ? amount.toFixed(2) : purchaseTotal.toFixed(2) }
```
Comment (smarcet) [inline src/components/order-summary.js:139]: same as https://github.com/fntechgit/summit-registration/pull/60/files#r714355968

File: src/components/order-summary.js [hunk 6]
```diff
@@ -217,7 +219,7 @@ class OrderSummary extends React.Component {
                           {T.translate("order_summary.total")}
                       </div>
                       <div className="col-xs-6 text-right total">
-                          ${ order.status === 'Paid' ? '0.00' : total.toFixed(2)}
+                          ${ order.status === 'Paid' ? '0.00' : reservation?.amount? reservation.amount.toFixed(2) : amount ? amount.toFixed(2) : purchaseTotal.toFixed(2)}
```
Comment (smarcet) [inline src/components/order-summary.js:222]: same as https://github.com/fntechgit/summit-registration/pull/60/files#r714355968

File: src/components/order-summary.js [hunk 7]
```diff
@@ -176,39 +168,49 @@ class OrderSummary extends React.Component {
...
                       </div>
                   </div>
                   }
+                  {reservation?.taxes_amount > 0 || taxes_amount > 0 && 
+                        <div className="row order-refunds order-row">
+                            <div className="col-xs-7 text-left">
+                                {T.translate("order_summary.taxes")}
+                            </div>
+                            <div className="col-xs-5 text-right subtotal">
+                                ${reservation?.taxes_amount? reservation.taxes_amount.toFixed(2) : taxes_amount?.toFixed(2)}
```
Comment (smarcet) [inline src/components/order-summary.js:202]: same as https://github.com/fntechgit/summit-registration/pull/60/files#r714355968

File: src/components/order-summary.js [hunk 8]
```diff
@@ -176,39 +168,49 @@ class OrderSummary extends React.Component {
...
                   }
-                  {refundTotal > 0 && 
+                  {reservation?.refunded_amount > 0 || refunded_amount > 0 && 
                   <div className="row order-refunds order-row">
                       <div className="col-xs-7 text-left">
                           {T.translate("order_summary.refunds")}                          
                       </div>
                       <div className="col-xs-5 text-right subtotal">
-                          -${refundTotal.toFixed(2)}
+                          -${reservation?.refunded_amount ? reservation.refunded_amount.toFixed(2) : refunded_amount?.toFixed(2)}
```
Comment (smarcet) [inline src/components/order-summary.js:192]: same as https://github.com/fntechgit/summit-registration/pull/60/files#r714355968

File: src/components/order-summary.js [hunk 9]
```diff
@@ -176,39 +168,49 @@ class OrderSummary extends React.Component {
...
                   })}
-                  {discountTotal > 0 &&
+                  {reservation?.discount_amount > 0 || discount_amount > 0 &&
                   <div className="row order-discounts order-row">
                       <div className="col-xs-7 text-left">
                           {T.translate("order_summary.discounts")}
                       </div>
                       <div className="col-xs-5 text-right subtotal">
-                          -${discountTotal.toFixed(2)}
+                          -${reservation?.discount_amount ? reservation.discount_amount.toFixed(2) : discount_amount?.toFixed(2)}
```
Comment (smarcet) [inline src/components/order-summary.js:182]: same as https://github.com/fntechgit/summit-registration/pull/60/files#r714355968

**Review by smarcet — CHANGES_REQUESTED** [review]
please review commnets

**Review by smarcet — APPROVED** [review]
looks good to me

**Pull request merged**